### PR TITLE
Refactor/react components

### PR DESCRIPTION
**Pull request by msakr21**
This is an initial round of refactoring.

I would like some input on how to organize the files in the components folder so it's intuitive and a bit more developer friendly, @bkeener7 .

### CODE REVIEW

**Comment by msakr21** [comment]
Please don't merge until you've taken a look at the components folder and we've had a chance to discuss how to organize it!

**Review by bkeener7 — APPROVED** [review]
We went through your changes together. Everything looks good on my end. I added some related commits, going to reverse Uno card the review back to you before merging.

File: spec/features/admin/enrollments/edit_spec.rb
Comment (msakr21) [inline spec/features/admin/enrollments/edit_spec.rb:8]: Absolutely love this!

File: spec/features/admin/enrollments/edit_spec.rb
Comment (msakr21) [inline spec/features/admin/enrollments/edit_spec.rb:12]: Same as above but I love it so much that here you go again:

File: spec/features/admin/enrollments/index_spec.rb
Comment (msakr21) [inline spec/features/admin/enrollments/index_spec.rb:15]: Great change!

**Pull request merged**
